### PR TITLE
Version tests

### PR DESCRIPTION
**Pull request by ben-bay**
Addresses issue #22.

Still figuring out how to add this to Travis in such a way that it runs only on `develop`, when it is seeking to merge into `master`.

Also, I'm still pretty bad at bash scripting so let me know if it can be improved.

### CODE REVIEW

File: tests/integration/version_test.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+# Ensure CHANGELOG version and merlin/__init__.py version are equal
+grep CHANGELOG.md -m 1 -e "## \[\d\+\.\d\+\.\d\+\]" > GREP1
+grep merlin/__init__.py -m 1 -e "\"\d\+\.\d\+\.\d\+\"" > GREP2
+grep -o GREP1 -e "\d\+\.\d\+\.\d\+" > GREP3
+grep -o GREP2 -e "\d\+\.\d\+\.\d\+" > GREP4
+if cmp GREP3 GREP4
```
Comment (koning) [inline tests/integration/version_test.sh]: You can use ! to negate the test to remove the  : and else.

Comment (ben-bay) [reply tests/integration/version_test.sh]: Fixed.

File: tests/integration/version_test.sh [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
+grep merlin/__init__.py -m 1 -e "\"[0-9]\+\.[0-9]\+\.[0-9]\+\"" > GREP2
+grep -o GREP1 -e "[0-9]\+\.[0-9]\+\.[0-9]\+" > GREP3
+grep -o GREP2 -e "[0-9]\+\.[0-9]\+\.[0-9]\+" > GREP4
+if ! cmp GREP3 GREP4
 then 
-    :
-else
     echo "Error: merlin/__init__.py version different from CHANGELOG.md verison" 
     exit 1
 fi
```
[the remark this answers is not in the document]
Comment (ben-bay) [reply tests/integration/version_test.sh]: > we also need to update docs/source/conf.py to pull the versions from merlin.VERSION

Isn't this already happening on line 21?
`MERLIN_VERSION = __import__("merlin").VERSION`

[the remark this answers is not in the document]
Comment (lucpeterson) [reply tests/integration/version_test.sh]: oh you're right it's fixed on develop: i was looking at master!

**Comment by ben-bay** [comment]
@lucpeterson I added in your suggestions, fixed a few bugs, and covered a few more corner cases.

**Comment by lucpeterson** [comment]
In terms of having this run only on merges into master I think you can change the .travis.yml file to use stages for only certain branches. like in this demo

https://github.com/travis-ci/build-stages-demo/blob/master/.travis.yml

**Pull request merged**